.github/workflows/stress.yml

```diff
@@ -43,66 +43,103 @@ env:
+    ${{
+      github.repository == 'temporalio/sdk-typescript'
+        && (github.ref == 'refs/heads/main'
+            || startsWith(github.ref, 'refs/tags/')
+            || startsWith(github.ref, 'refs/heads/releases'))
+        && github.event_name != 'pull_request'
+    }}
 
 jobs:
   stress-test:
     runs-on: buildjet-4vcpu-ubuntu-2204
     steps:
       - name: Print build info
         run: 'echo test-type: ${{ inputs.test-type }}, test-timeout-minutes: ${{ inputs.test-timeout-minutes }}, reuse-v8-context: $REUSE_V8_CONTEXT'
-      - uses: actions/checkout@v2
+
+      - uses: actions/checkout@v4
         with:
           submodules: recursive
-      - uses: actions-rs/toolchain@v1
+
+      - name: Install Node
+        uses: actions/setup-node@v3
         with:
-          toolchain: stable
-      - uses: actions/setup-node@v1
+          node-version: 20
+
+      - name: Get NPM cache directory
+        id: npm-cache-dir
+        run: echo "dir=$(npm config get cache)" >> ${GITHUB_OUTPUT}
+
+      - name: Restore NPM cache
+        uses: actions/cache/restore@v3
         with:
-          node-version: ${{ matrix.node }}
+          path: ${{ steps.npm-cache-dir.outputs.dir }}
+          key: npm-main-linux-intel-${{ hashFiles('./package-lock.json') }}
+          restore-keys: |
+            npm-main-linux-intel-
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          target: x86_64-unknown-linux-gnu
+
       - name: Install protoc
-        uses: arduino/setup-protoc@v1
+        uses: arduino/setup-protoc@v3
         with:
-          version: '3.x'
+          # TODO: Upgrade proto once https://github.com/arduino/setup-protoc/issues/99 is fixed
+          version: '23.x'
           repo-token: ${{ secrets.GITHUB_TOKEN }}
-      - uses: Swatinem/rust-cache@v1
+
+      - name: Rust Cargo and Build cache
+        uses: Swatinem/rust-cache@v2
         with:
-          working-directory: packages/core-bridge
-      # Don't build during install phase since we're going to explicitly build
-      - run: npm ci --ignore-scripts
-      - run: npm run build
+          workspaces: packages/core-bridge -> target
+          prefix-key: corebridge-buildcache
+          shared-key: linux-intel
+          env-vars: ''
+          save-if: ${{ env.IS_MAIN_BRANCH }}
+
+      - name: Download dependencies
+        # Make up to 3 attempts to install NPM dependencies, to work around transient NPM errors
+        run: |
+          npm ci --ignore-scripts --verbose || npm ci --ignore-scripts --verbose || npm ci --ignore-scripts --verbose
+
+      - name: Compile code
+        run: npm run build
         env:
           BUILD_CORE_RELEASE: true
 
-      # docker-compose
-      - name: Checkout docker-compose repo
-        uses: actions/checkout@v2
-        with:
-          repository: temporalio/docker-compose
-          path: docker-compose
+      - name: Install Temporal CLI
+        uses: temporalio/setup-temporal@v0
```

Comment:
Is this action even stable?

Reply:
It was about a year ago, and I'm not aware of any problem with it ATM. It goes through `https://temporal.download/cli/`, not the shell script, and clearly, still work properly now that CLI 0.12.0 has been released.

But agree we should get some more pair of eyes on that repo.